### PR TITLE
Moved netpbm to a safe installation source

### LINKED ISSUE #4420: Update recipe for netpbm to use a "safe" source
netpbm uses an unsafe source and wont be build or show up in HaikuDepot, update the recipe to use the latest release and make the needed changes to the recipe.
Current recipe is found in media-libs/netpbm

### PR DESCRIPTION
**Pull request by redsPL**
Fixed #4420

### CODE REVIEW

**Comment by korli** [comment]
Please fetch from GitHub mirror.

**Comment by redsPL** [comment]
@korli How do we know that the GitHub mirror is official and legitimate? `t6` user isn't listed anywhere on their sourceforge page, so by using the GH miror we could introduce a possible security threat to the Haiku infrastructure.

**Comment by pulkomandy** [comment]
Why are we not using the much more recent release from https://sourceforge.net/projects/netpbm/files/super_stable/10.73.29/ instead?

**Comment by redsPL** [comment]
Mainly because I'm blind. I'm currently building the new package, if everything works as expected I'll submit another commit.

**Comment by redsPL** [comment]
The new version builds OK. Sorry for the double commit, I'm still not the best git user :c

File: media-libs/netpbm/netpbm-10.73.29.recipe [hunk 1]
```diff
@@ -231,9 +231,9 @@ COPYRIGHT="2001 411toppm: Steve Allen
 	1993 zeisstopnm: Oliver Trepte"
 LICENSE="GNU GPL v2"
 REVISION="2"
-SOURCE_URI="svn+http://svn.code.sf.net/p/netpbm/code/advanced#2294"
-CHECKSUM_SHA256="835ce1d42ea9f50eddf74754f2b06b1c0f7a1d8e46deb89b839a5ca018599793"
-SOURCE_DIR="advanced"
+SOURCE_URI="https://downloads.sourceforge.net/project/netpbm/super_stable/10.73.29/netpbm-10.73.29.tgz"
+CHECKSUM_SHA256="e4871805d3be46b98047a30774186da3fef0855ee4fe31eb9c16c5e5cffa96d9"
+SOURCE_DIR="netpbm-10.73.29/"
```
Comment (Begasus) [inline media-libs/netpbm/netpbm-10.73.29.recipe]: Try to use $portVersion variable:
```diff
-SOURCE_URI="https://downloads.sourceforge.net/project/netpbm/super_stable/10.73.29/netpbm-10.73.29.tgz"
CHECKSUM_SHA256="e4871805d3be46b98047a30774186da3fef0855ee4fe31eb9c16c5e5cffa96d9"
-SOURCE_DIR="netpbm-10.73.29/"
+SOURCE_URI="https://downloads.sourceforge.net/project/netpbm/super_stable/$portVersion/netpbm-$portVersion.tgz"
CHECKSUM_SHA256="e4871805d3be46b98047a30774186da3fef0855ee4fe31eb9c16c5e5cffa96d9"
+SOURCE_DIR="netpbm-$portVersion"
```

**Comment by Begasus** [comment]
Maybe we could move the cmd's to a seperate package _tools?

**Comment by korli** [comment]
well the latest version is 10.88.01, see https://sourceforge.net/p/netpbm/code/3711/
the github mirror is used by https://svnweb.freebsd.org/ports/head/graphics/netpbm/
see https://svnweb.freebsd.org/ports?view=revision&revision=451378

**Comment by redsPL** [comment]
@Begasus I'll change that version tommorow, thanks for the tip.
@korli In that case, if the latest version will build properly, I'll update it even further. Thanks!

**Comment by Begasus** [comment]
Repology reports 10.88.01 as the latest version: https://repology.org/project/netpbm/versions
But many still use the older version.

**Comment by Begasus** [comment]
@korli what do you think, should we leave the cmd's as is or create a _tools package for it?
So far nothing is using lib:libnetpbm, only the cmd's are really useful atm I think, so maybe leave them in?

**Comment by korli** [comment]
I would leave as is, the package wasn't built until then.

**Comment by redsPL** [comment]
I managed to fix the recipe errors, but the build still crashes on libxml. Turns out that Haiku's libxml2 package doesn't provide xml2-config binary..? Should we just package an older version for now?
```
(...)
Package libxml-2.0 was not found in the pkg-config search path.
Perhaps you should add the directory containing `libxml-2.0.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libxml-2.0' found
make: xml2-config: Command not found
Package x11 was not found in the pkg-config search path.
Perhaps you should add the directory containing `x11.pc'
to the PKG_CONFIG_PATH environment variable
No package 'x11' found
(...)
```

**Comment by Begasus** [comment]
Do you have the libxml2_devel package installed? It should be in there: https://github.com/haikuports/haikuports/blob/cf8efc9f2037426b64b35a9f6fcc3bf59cfd4089/dev-libs/libxml2/libxml2-2.9.10.recipe#L46

**Comment by Begasus** [comment]
Also, it's missing in the recipe "lib:libxml2" and "devel:libxml2"

**Comment by redsPL** [comment]
libxml2_devel doesn't appear to be present in HaikuDepot..? Also, it doesn't properly include if I add it to netbpm's dependency list.

**Comment by Begasus** [comment]
Try searching (in Terminal) with "pkgman search libxml2"

**Comment by redsPL** [comment]
It finally builds! Had some problems, but they're all fixed now. I hope that it'll be OK..?

**Comment by redsPL** [comment]
@pulkomandy oh, so I actually misread @korli's message and thought that I should build the latest-latest version. Should I, like, revert those changes and just go back to 10.73.29?

**Comment by pulkomandy** [comment]
No, I didn't see korli's message saying that freebsd is also using the newer version. I think that makes it official enough, and you spent enough effort on this already :)

File: media-libs/netpbm/netpbm-10.88.01.recipe [hunk 2]
```diff
@@ -623,6 +648,8 @@ BUILD_REQUIRES="
 	devel:libpng$secondaryArchSuffix
 	devel:libtiff$secondaryArchSuffix
 	devel:libz$secondaryArchSuffix
+	devel:libx11$secondaryArchSuffix
```
Comment (Begasus) [inline media-libs/netpbm/netpbm-10.88.01.recipe]: Not needed:
```diff
-	devel:libx11$secondaryArchSuffix
```

File: media-libs/netpbm/netpbm-10.88.01.recipe [hunk 3]
```diff
@@ -606,11 +626,16 @@ REQUIRES="
 	lib:libpng$secondaryArchSuffix
 	lib:libtiff$secondaryArchSuffix
 	lib:libz$secondaryArchSuffix
+	lib:libxml2$secondaryArchSuffix
+	devel:libxml2$secondaryArchSuffix
+	cmd:xml2_config
+	lib:libx11$secondaryArchSuffix
```
Comment (Begasus) [inline media-libs/netpbm/netpbm-10.88.01.recipe]: Not needed:
```diff
-	devel:libxml2$secondaryArchSuffix
-	cmd:xml2_config
-	lib:libx11$secondaryArchSuffix
```

File: media-libs/netpbm/netpbm-10.88.01.recipe [hunk 4]
```diff
@@ -606,11 +626,16 @@ REQUIRES="
 	lib:libpng$secondaryArchSuffix
 	lib:libtiff$secondaryArchSuffix
 	lib:libz$secondaryArchSuffix
+	lib:libxml2$secondaryArchSuffix
```
Comment (Begasus) [inline media-libs/netpbm/netpbm-10.88.01.recipe]: Trade places, libx comes before libz

File: media-libs/netpbm/netpbm-10.88.01.recipe [hunk 5]
```diff
@@ -606,11 +626,16 @@ REQUIRES="
...
+	devel:libxml2$secondaryArchSuffix
+	cmd:xml2_config
+	lib:libx11$secondaryArchSuffix
 	"
 
 PROVIDES_devel="
 	netpbm${secondaryArchSuffix}_devel = $portVersion compat >= 10
-	devel:libnetpbm$secondaryArchSuffix = 11.68 compat >= 11
+	libnetpbm$secondaryArchSuffix = 11.68 compat >= 11
+	devel:libnetpbm
```
Comment (Begasus) [inline media-libs/netpbm/netpbm-10.88.01.recipe]: Not needed:
```diff
-	libnetpbm$secondaryArchSuffix = 11.68 compat >= 11
-	devel:libnetpbm
```

File: media-libs/netpbm/netpbm-10.88.01.recipe [hunk 6]
```diff
@@ -606,11 +626,16 @@ REQUIRES="
...
 	lib:libz$secondaryArchSuffix
+	lib:libxml2$secondaryArchSuffix
+	devel:libxml2$secondaryArchSuffix
+	cmd:xml2_config
+	lib:libx11$secondaryArchSuffix
 	"
 
 PROVIDES_devel="
 	netpbm${secondaryArchSuffix}_devel = $portVersion compat >= 10
-	devel:libnetpbm$secondaryArchSuffix = 11.68 compat >= 11
```
Comment (Begasus) [inline media-libs/netpbm/netpbm-10.88.01.recipe]: libversion is 11.88 now

File: media-libs/netpbm/netpbm-10.88.01.recipe [hunk 7]
```diff
@@ -596,6 +615,7 @@ PROVIDES="
 	cmd:ybmtopbm
 	cmd:yuvsplittoppm
 	cmd:yuvtoppm
+	cmd:yuy2topam
 	cmd:zeisstopnm
 	lib:libnetpbm$secondaryArchSuffix = 11.68 compat >= 11
```
Comment (Begasus) [inline media-libs/netpbm/netpbm-10.88.01.recipe]: libversion is 11.88 now

File: media-libs/netpbm/netpbm-10.88.01.recipe [hunk 8]
```diff
@@ -623,6 +648,8 @@ BUILD_REQUIRES="
 	devel:libpng$secondaryArchSuffix
 	devel:libtiff$secondaryArchSuffix
```
Comment (Begasus) [inline media-libs/netpbm/netpbm-10.88.01.recipe]: add devel:libxml2$secondaryArchSuffix here
```diff
+	devel:libxml2$secondaryArchSuffix
```

File: media-libs/netpbm/netpbm-10.88.01.recipe [hunk 9]
```diff
@@ -631,6 +658,7 @@ BUILD_PREREQUIRES="
 	cmd:make
 	cmd:perl
 	cmd:pkg_config$secondaryArchSuffix
+	cmd:xml2_config
```
Comment (Begasus) [inline media-libs/netpbm/netpbm-10.88.01.recipe]: Not needed (provided by devel:libxml2)
```diff
-	cmd:xml2_config
```

File: media-libs/netpbm/patches/netpbm-10.88.01.patchset [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+diff --git a/lib/Makefile b/lib/Makefile
+index bc758df..e829247 100644
+--- a/lib/Makefile
++++ b/lib/Makefile
+@@ -131,7 +131,7 @@ $(SONAME): libnetpbm.$(NETPBMLIBSUFFIX).$(MAJ).$(MIN)
+ 	$(SYMLINK) $< $@
+ libnetpbm.$(NETPBMLIBSUFFIX).$(MAJ).$(MIN): $(LIBOBJECTS) $(LIBOBJECTS_X)
+ 	$(LD) $(LDSHLIB) -o $@ $(LIBOBJECTS) $(LIBOBJECTS_X) \
+-          $(SHLIB_CLIB) -lm $(LADD)
++          $(SHLIB_CLIB) $(MATHLIB) $(LADD)
```
Comment (Begasus) [inline media-libs/netpbm/patches/netpbm-10.88.01.patchset]: This isn't actually required anymore, Haiku provides libm

File: media-libs/netpbm/netpbm-10.88.01.recipe (hunk 5 above)
Comment (redsPL) [reply media-libs/netpbm/netpbm-10.88.01.recipe]: removing `devel:libnetpbm` actually breaks the build for me

**Comment by redsPL** [comment]
@Begasus gcc2 build crashes for me on the following:
```
In file included from /boot/system/develop/headers/os/support/ByteOrder.h:11,
                 from /boot/system/develop/headers/bsd/endian.h:16,
                 from /boot/system/develop/headers/posix/netinet/in.h:10,
                 from /boot/home/haikuports/media-libs/netpbm/work-10.88.01/sources/netpbm-10.88.01/converter/ppm/ppmtompeg/headers/byteorder.h:76,
                 from /boot/home/haikuports/media-libs/netpbm/work-10.88.01/sources/netpbm-10.88.01/converter/ppm/ppmtompeg/bitio.c:50:
/boot/system/develop/headers/os/support/SupportDefs.h:20: warning: redefinition of `int8'
/boot/home/haikuports/media-libs/netpbm/work-10.88.01/sources/netpbm-10.88.01/converter/ppm/ppmtompeg/headers/general.h:121: warning: `int8' previously declared here
/boot/system/develop/headers/os/support/SupportDefs.h:21: warning: redefinition of `uint8'
/boot/home/haikuports/media-libs/netpbm/work-10.88.01/sources/netpbm-10.88.01/converter/ppm/ppmtompeg/headers/general.h:120: warning: `uint8' previously declared here
/boot/system/develop/headers/os/support/SupportDefs.h:22: warning: redefinition of `int16'
/boot/home/haikuports/media-libs/netpbm/work-10.88.01/sources/netpbm-10.88.01/converter/ppm/ppmtompeg/headers/general.h:123: warning: `int16' previously declared here
/boot/system/develop/headers/os/support/SupportDefs.h:23: warning: redefinition of `uint16'
/boot/home/haikuports/media-libs/netpbm/work-10.88.01/sources/netpbm-10.88.01/converter/ppm/ppmtompeg/headers/general.h:122: warning: `uint16' previously declared here
/boot/system/develop/headers/os/support/SupportDefs.h:24: conflicting types for `int32'
/boot/home/haikuports/media-libs/netpbm/work-10.88.01/sources/netpbm-10.88.01/converter/ppm/ppmtompeg/headers/general.h:150: previous declaration of `int32'
/boot/system/develop/headers/os/support/SupportDefs.h:25: conflicting types for `uint32'
/boot/home/haikuports/media-libs/netpbm/work-10.88.01/sources/netpbm-10.88.01/converter/ppm/ppmtompeg/headers/general.h:149: previous declaration of `uint32'
/boot/home/haikuports/media-libs/netpbm/work-10.88.01/sources/netpbm-10.88.01/common.mk:269: recipe for target 'bitio.o' failed
```

Should we attempt to fix those, or just call it a day and mark it as broken for x86_gcc2? If the earlier, I probably won't submit a patch until ~Sunday because I have a lot of things going on IRL.

**Comment by pulkomandy** [comment]
As long as it builds with gcc8, you're fine. If gcc2 support is needed itt's fine to keep an old version of the software for that purpose.

**Comment by redsPL** [comment]
OK, that should be everything. Can we merge?

File: media-libs/netpbm/netpbm-10.88.01.recipe (hunk 5 above)
Comment (Begasus) [reply media-libs/netpbm/netpbm-10.88.01.recipe]: My bad on that one :)
Should be:
```diff
-	devel:libnetpbm
+	devel:libnetpbm$secondaryArchSuffix = 11.88 compat >= 11
```

File: media-libs/netpbm/netpbm-10.88.01.recipe (hunk 6 above)
Comment (Begasus) [reply media-libs/netpbm/netpbm-10.88.01.recipe]: Check the lib version in the package (or in the source after build), it's **libnetpbm.so.11.88**

**Comment by Begasus** [comment]
Commented on a few things, but it's looking good already!
For the error on the int types, it's a 32bit problem, fix should be easy (check for a way to patch the general.h file)

**Comment by Begasus** [comment]
@korli in the previous patch there is one to include stdbool.h, doesn't error out on it, would it be safe to leave it out?

**Comment by korli** [comment]
@Begasus I can't say, sorry. You can always check after merging.

**Comment by redsPL** [comment]
Fiddled with gcc2 a bit more, but had no luck with building. Currently marked the package as broken for x86_gcc2, I might try to fix it in the future.

**Comment by Begasus** [comment]
> Fiddled with gcc2 a bit more, but had no luck with building. Currently marked the package as broken for x86_gcc2, I might try to fix it in the future.
Check this for a hint (it's actually quite simple) :) https://github.com/haikuports/haikuports/blob/bccd6fba1157df864b9d00a2592f03c386e686d7/games-emulation/advancemame/patches/advancemame_x86-3.9.patchset#L15

**Comment by redsPL** [comment]
Well, thanks @Begasus, that helped a lot!

I hope that's everything now

**Comment by Begasus** [comment]
Could you adjust the libversion? Already left a comment earlier about it :)

File: media-libs/netpbm/patches/netpbm-10.88.01.patchset [hunk 11]
```diff
@@ -0,0 +1,48 @@
+From 138a84d8b3702b5dcfac726a32d6880498f6c8d4 Mon Sep 17 00:00:00 2001
+From: reds <reds@sakamoto.pl>
+Date: Sat, 14 Dec 2019 20:30:06 +0100
+Subject: weeeeeeee it works
```
Comment (Begasus) [inline media-libs/netpbm/patches/netpbm-10.88.01.patchset]: Nice subject but doesn't really say what's it's doing :)

**Comment by Begasus** [comment]
@korli had a look again for gcc2h, could we drop the cmd's for secondary architecture and just keep the libraries in there? Don't think having the secondaryArchSuffix binaries would bring much more if they are available for primary architecture (@redsPL don't bother for this atm) :)

File: media-libs/netpbm/netpbm-10.88.01.recipe [hunk 12]
```diff
@@ -659,16 +681,13 @@ INSTALL()
...
 	mv $prefix/temp/include/* $developDir/headers/
-	rm -r $manDir/web
-	rm $binDir/manweb $binDir/doc.url
 	rm $prefix/temp/{README,VERSION,config_template,pkgconfig_template,pkginfo}
 	mv $prefix/temp/misc $dataDir/netpbm
 	rm -r $prefix/temp
 	ln -s libnetpbm.so.11 $libDir/libnetpbm.so
 
-	prepareInstalledDevelLibs libnetpbm
+	prepareInstalledDevelLibs libnetpbm$secondaryArchSuffix
```
Comment (Begasus) [inline media-libs/netpbm/netpbm-10.88.01.recipe]: $secondaryArchSuffix not needed here:
```diff
-prepareInstalledDevelLibs libnetpbm$secondaryArchSuffix
+prepareInstalledDevelLibs libnetpbm
```

**Comment by Begasus** [comment]
Thanks for the contribution @redsPL , just pushed a tiny change to be able to merge it :)

**Pull request merged**